src/Infrastructure/Processing/AssetProcessingWorkerLoop.php

```
@@ -80,15 +80,25 @@ private function processReservation(ReservedAssetProcessingJob $reservation): vo
+            AssetProcessingJobDelivery::DEAD_LETTER => $reservation->deadLetter($this->requiredQueuePayload($result)),
             AssetProcessingJobDelivery::HANDLED => $reservation->acknowledge(),
             AssetProcessingJobDelivery::DISCARD => $reservation->discard(),
-            AssetProcessingJobDelivery::RETRY => $reservation->release(),
+            AssetProcessingJobDelivery::RETRY => $reservation->release($this->requiredQueuePayload($result)),
         };
     }
 
+    private function requiredQueuePayload(AssetProcessingJobHandlingResult $result): string
+    {
+        if ($result->queuePayload === null) {
+            throw new \LogicException('Queue payload is required for retry or dead-letter deliveries.');
```

PHPMD (2.15.0)

[error] Missing class import via use statement (line '96', column '23'). (undefined)

(MissingImport)